### PR TITLE
Update date for dropping Java 8 support in weekly line

### PR DESCRIPTION
**Pull request by timja**
See https://github.com/jenkinsci/jep/pull/387

image

### Proposed changelog entries

* Announce Java 8 end of life for weekly line as June 21 2022

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

### CODE REVIEW

**Review by daniel-beck — CHANGES_REQUESTED** [review]
Changing the date so significantly is hostile to users who already saw the current message.

At a minimum, this needs a good justification other than "we never meant the previous date", as well as another change to the ID to ensure this shows up for everyone again.

**Review by basil — APPROVED** [review]
Messaging could have been smoother, but at this point the important thing is to get it done.

**Comment by MarkEWaite** [comment]
> Outsch. This will rapidly change our plans. Currently we do our best to migrate our jenkins instance in to new server. But I think we will be not finished before 1. Of July. Is there any reason why is the date changed?
> FYI our currently instance runs on some old CentOS where is not possible upgrade java to 11. And we need also upgrade all of our 300 clients.

I assume that you're running the long term support release rather than a weekly release in that environment.  If so, then this is no change from the previously published warning from the administrative monitor.  LTS will be supported on Java 8 until Sep 2022.

If you're running Jenkins weekly in that environment, you could consider switching from weekly to LTS in order to have more time to run Jenkins on Java 8.

If you're running on CentOS 6 or older, then you're already outside the Jenkins support policy.  Our [Linux support policy](https://www.jenkins.io/doc/administration/requirements/linux/) states we don't support Jenkins on operating systems that are not supported by their operating system provider.  I don't think we can reasonably support Jenkins on an operating system that is past the end of its life.

File: core/src/main/resources/jenkins/monitor/JavaVersionRecommendationAdminMonitor/message.properties [hunk 1]
```diff
@@ -1,2 +1,2 @@
-Recommended_Java_Version=You are running Jenkins on Java {0}, support for which will end on or after September 1, 2022. \
+Recommended_Java_Version=You are running Jenkins on Java {0}, support for which will end on or after June 21, 2022. \
```
[the remark this answers is not in the document]
Comment (daniel-beck) [reply core/src/main/resources/jenkins/monitor/JavaVersionRecommendationAdminMonitor/message.properties:1]: Should also acknowledge/emphasize that the date changed from before. People are not going to read the "same" message again.

(Ideally the message would be different based on LTS line but that's probably more trouble than it's worth.)

**Comment by mPokornyETM** [comment]
Thanks too much.  We can switch to lts in worst case to have time for our development.
 This is good idea.
Later we will run in the official jenkins docker release.  Our update strategy cab be fully automated now and I hope we spare maintenence time in my team.

File: core/src/main/resources/jenkins/monitor/JavaVersionRecommendationAdminMonitor/message.properties (hunk 1 above)
[the remark this answers is not in the document]
Comment (timja) [reply core/src/main/resources/jenkins/monitor/JavaVersionRecommendationAdminMonitor/message.properties:1]: Any wording suggestion?

**Comment by timja** [comment]
I've added emphasis to the message, I think it looks a lot better now:
image

File: core/src/main/resources/jenkins/monitor/JavaVersionRecommendationAdminMonitor/message.properties (hunk 1 above)
[the remark this answers is not in the document]
Comment (timja) [reply core/src/main/resources/jenkins/monitor/JavaVersionRecommendationAdminMonitor/message.properties]: Does this emphasis address your concerns?
https://github.com/jenkinsci/jenkins/pull/6566#issuecomment-1126985151

or have you got any other suggestions?

**Review by basil — APPROVED** [review]
Thank you Tim, this looks really great and is much fancier than anything I could have come up with.

File: core/src/main/resources/jenkins/monitor/JavaVersionRecommendationAdminMonitor/message.properties (hunk 1 above)
[the remark this answers is not in the document]
Comment (daniel-beck) [reply core/src/main/resources/jenkins/monitor/JavaVersionRecommendationAdminMonitor/message.properties]: "This is a change from a previously announced date." or similar. I did not mean formatting when I wrote emphasis, sorry for the confusion.

**Comment by daniel-beck** [comment]
Do we plan to include the same updated message in LTS, or only weekly releases?

**Comment by timja** [comment]
> Do we plan to include the same updated message in LTS, or only weekly releases?

I was thinking about doing something similar, but wondering whether we wait to .2 to add the emphasis.

but open to doing something similar now as well.

File: core/src/main/resources/jenkins/monitor/JavaVersionRecommendationAdminMonitor/message.properties (hunk 1 above)
[the remark this answers is not in the document]
Comment (timja) [reply core/src/main/resources/jenkins/monitor/JavaVersionRecommendationAdminMonitor/message.properties]: added, what do you think?

image

**Comment by timja** [comment]
I've put the weekly release on hold for now so as to integrate this.

Any feedback here please @daniel-beck / @jenkinsci/core ?

**Review by daniel-beck — APPROVED** [review]
Implementation-wise, this looks fine. Thanks for making the adjustments.

**Comment by timja** [comment]
(expedited for weekly to give as much notice as possible)

**Comment by MarkEWaite** [comment]
Apologies for delayed approval.  This looks great to me.  Sorry for my original mistake approving the declaration of the end of life date as September 2022 without stating my assumption that was for the LTS release.  Failure on my part by not being clear on my assumptions.

**Pull request merged**